src/main/java/redis/clients/jedis/ShardedJedisPool.java

```diff
@@ -60,17 +72,31 @@ protected void returnResource(final ShardedJedis resource) {
     private List<JedisShardInfo> shards;
     private Hashing algo;
     private Pattern keyTagPattern;
+    private String clientName;
 
-    public ShardedJedisFactory(List<JedisShardInfo> shards, Hashing algo, Pattern keyTagPattern) {
+    public ShardedJedisFactory(List<JedisShardInfo> shards, Hashing algo, Pattern keyTagPattern, String clientName) {
       this.shards = shards;
       this.algo = algo;
       this.keyTagPattern = keyTagPattern;
+      this.clientName = clientName;
     }
 
     @Override
     public PooledObject<ShardedJedis> makeObject() throws Exception {
-      ShardedJedis jedis = new ShardedJedis(shards, algo, keyTagPattern);
-      return new DefaultPooledObject<ShardedJedis>(jedis);
+      ShardedJedis shardedJedis = new ShardedJedis(shards, algo, keyTagPattern);
+      for (Jedis jedis : shardedJedis.getAllShards()) {
+        try{
+          jedis.connect();
+          if (clientName != null) {
+            jedis.clientSetname(clientName);
+          }
+        } catch (JedisException je) {
+          log.log(Level.SEVERE, "Can't connect to redis at " + jedis.getClient().getHost(), je);
+          jedis.close();
```

Comment:
raise exception will cause this case:  one redis server's connection fail will cause we can't work even other redis servers is Ok. Am I right? So I swallow it to make sure makeobject always success.